fastplotlib/graphics/selectors/_rectangle.py

```diff
@@ -0,0 +1,511 @@
+                                if len(missing_ixs) > 0:
+                                    data = g.data[s].copy()
+                                    data[missing_ixs] = np.nan
+                                    data_selections.append(data)
+                                else:
+                                    data_selections.append(g.data[s])
+                            case "ignore":
+                                if len(missing_ixs) > 0:
+                                    data_selections.append(
+                                        np.array([], dtype=np.float32).reshape(0, 3)
+                                    )
+                                else:
+                                    data_selections.append(g.data[s])
+                return data_selections
+            else:
+                if ixs.size == 0:
+                    # empty selection
+                    return np.array([], dtype=np.float32).reshape(0, 3)
+
+                s = slice(
+                    ixs[0], ixs[-1] + 1
+                )  # add 1 to end because these are direct indices
+                # slices n_datapoints dim
+                # slice with min, max is faster than using all the indices
+
+                # get missing ixs
+                missing_ixs = np.setdiff1d(np.arange(ixs[0], ixs[-1] + 1), ixs) - ixs[0]
+
+                match mode:
+                    case "full":
+                        return source.data[s]
+                    case "partial":
+                        if len(missing_ixs) > 0:
+                            data = source.data[s].copy()
+                            data[missing_ixs] = np.nan
+                            return data
+                        else:
+                            return source.data[s]
+                    case "ignore":
+                        if len(missing_ixs) > 0:
+                            warnings.warn(
+                                "You have selected 'ignore' mode. Selected graphic has incomplete indices. "
+                                "Move the selector or change the mode to one of `partial` or `full`."
+                            )
+                            return np.array([], dtype=np.float32)
+                        else:
+                            return source.data[s]
+
+    def get_selected_indices(
+        self, graphic: Graphic = None
+    ) -> Union[np.ndarray, List[np.ndarray]]:
+        """
+        Returns the indices of the ``Graphic`` data bounded by the current selection.
+
+        These are the data indices which correspond to the data under the selector.
+
+        Parameters
+        ----------
+        graphic: Graphic, default ``None``
+            If provided, returns the selection indices from this graphic instrad of the graphic set as ``parent``
+
+        Returns
+        -------
+        Union[np.ndarray, List[np.ndarray]]
+            data indicies of the selection, list of np.ndarray if the graphic is a collection
```

Comment:
For image is it `list[x_indices_array, y_indices_array]` and for line/line collection what is the shape =S ? Can specify here